src/Navigation/Drawer/DrawerStyle.ts

```diff
@@ -44,39 +44,51 @@ const closeToLeft = keyframes`
 const fadeIn = keyframes`
   from {
     opacity: 0;
+    visibility: hidden;
+    background-color: transparent;
   }
-
+  
   to {
     opacity: 1;
+    visibility: visible;
+    background-color: rgba(0, 0, 0, 0.5);
   }
-`;
+  `;
 
 const fadeOut = keyframes`
   from {
     opacity: 1;
+    visibility: visible;
+    background-color: rgba(0, 0, 0, 0.5);
   }
-
+  
   to {
     opacity: 0;
+    visibility: hidden;
+    background-color: transparent;
   }
 `;
 
 export const DrawerContainer = styled.div<DrawerContainerProps>`
   position: fixed;
-  display: ${({ isDisplay }) => (isDisplay ? 'block' : 'none')};
   min-height: 100vh;
   max-height: 100vh;
   width: 100vw;
   top: 0px;
   left: 0px;
-  background-color: rgba(0, 0, 0, 0.5);
-  animation: ${({ open }) => (open ? fadeIn : fadeOut)} 0.3s ease-in-out;
-  z-index: ${({ isDisplay }) => (isDisplay ? '9999' : '-1')};
+  opacity: ${({ open }) => (open ? '1' : '0')};
+  visibility: ${({ open }) => (open ? 'visible' : 'hidden')};
```

Comment:
Why not use the display property to control this?

Reply:
`display` can not be animated.

Reply:
Oh it was being used earlier, I thought it should be working following the same styles. Also had tried it on my local, could see the fade in animation.

Reply:
The previous implementation did it roughly like this: Changing the `open` prop triggered the animation. And the start/end of that animation then triggered the changing of the `isDisplay` state, which triggered the `display` css prop to change. It was important to get the timing right for the `display` prop because you can't set `display` to `none` while the animation is still running. But with `visibility` combined with `transition` this is possible.